src/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { FC } from 'react'
+
+import { useTranslation } from 'react-i18next'
+import FilterListIcon from '@mui/icons-material/FilterList'
+import Typography from '@mui/material/Typography'
+import Button from '@mui/material/Button'
+import { Box } from '@mui/material'
+
+import AppDrawer from '~/components/app-drawer/AppDrawer'
+import AppSelect from '~/components/app-select/AppSelect'
+import CheckboxList from '~/components/checkbox-list/CheckboxList'
+
+import { spliceSx } from '~/utils/helper-functions'
+import { styles } from '~/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer.styles'
+import {
+  ButtonVariantEnum,
+  CourseFilters,
+  ProficiencyLevelEnum,
+  TypographyVariantEnum
+} from '~/types'
+
+interface CoursesFiltersDrawerProps {
+  filters: CourseFilters
+  handleFilterChange: (
+    key: keyof CourseFilters,
+    value: string | ProficiencyLevelEnum[]
+  ) => void
+  onClose: () => void
+  open: boolean
+}
+const fields = [
+  { value: 'music', title: 'Music' },
+  { value: 'lang', title: 'Languages' },
+  { value: 'development', title: 'Development' }
+]
+
+const CoursesFiltersDrawer: FC<CoursesFiltersDrawerProps> = ({
+  filters,
+  handleFilterChange,
+  onClose,
+  open
+}) => {
+  const levelOptions = Object.values(ProficiencyLevelEnum)
+  const { t } = useTranslation()
+  const clearFilters = () => {
+    handleFilterChange('category', '')
+    handleFilterChange('subject', '')
+    handleFilterChange('proficiencyLevel', [])
+  }
+
+  return (
+    <AppDrawer
+      PaperProps={{ sx: styles.paper }}
+      anchor='left'
+      onClose={onClose}
+      open={open}
+    >
+      <Box sx={styles.titleWithIcon}>
+        <FilterListIcon sx={styles.icon} />
+        <Typography sx={styles.title} variant={TypographyVariantEnum.H6}>
+          {t('myCoursesPage.coursesFilter.coursesFilterLabel')}
+        </Typography>
+      </Box>
+
+      <Box sx={styles.categorySelect}>
+        <Typography sx={styles.titleMargin}>
+          {t('myCoursesPage.coursesFilter.chooseThe')}
+          <Typography sx={styles.boldText}>
+            {t('myCoursesPage.coursesFilter.category')}:
+          </Typography>
+        </Typography>
+        <AppSelect
+          fields={fields}
+          fullWidth
+          label='Category'
+          setValue={(value) => handleFilterChange('category', value)}
+          size='small'
+          value={filters.category}
+        />
+      </Box>
+
+      <Box sx={styles.subjectSelect}>
+        <Typography
+          sx={
+            !filters.category
+              ? spliceSx(styles.disabledTitle, styles.titleMargin)
+              : styles.titleMargin
+          }
+        >
+          {t('myCoursesPage.coursesFilter.chooseThe')}
+          <Typography
+            sx={!filters.category ? styles.inlineBlock : styles.boldText}
```

Comment:
`sx={styles.inlineBlock(!!filters.category)} `
And in styles set fontWeight: 500 if true